src/simple_websocket/ws.py

```diff
@@ -163,11 +163,15 @@ def _thread(self):
                 if len(in_data) == 0:
                     raise OSError()
             except (OSError, ConnectionResetError):  # pragma: no cover
+                self.close(reason=CloseReason.GOING_AWAY,
+                           message='Connection Error')
```

Comment:
Maybe there is some corner case that I'm missing, but isn't this exception handler being invoked when the other side goes away? So what would be the logic to send a close packet to that other side that is already gone?

Reply:
I left it there because I couldn't be sure that an OSError was caused only by the client going away, it could be something on our side (system issues), for example problems in the recv(). Replacing it with a self.sock.close() would work, however we may incur in protocol violation if it's not true that the other side went away, I reckon.

Reply:
Reanimating this: Should I change this to self.sock.close()? @miguelgrinberg